### ISSUE TITLE
Assign unique subscriber number for users that become monthly subscribers

### ISSUE THREAD

**Issue by malqinneh** [mdn/sprints#2989]
# User story
As a monthly subscriber, I want a unique subscriber number, so I can feel good about supporting MDN

# Design
- Link goes here

# Acceptance criteria

- [ ] As a user considering to subscribe monthly, I can see the unique subscriber number on the [FAQ page] and [account settings] that I will become if I sign up
- [ ] As a user that subscribed monthly, I can see my unique subscriber number on the [Thank You page] and my [account settings]

**Comment by chinikes** [comment]
Hi @tobinmori - This story needs the Team's analysis and tasking out before planning tomorrow. Thanks!

**Comment by tobinmori** [comment]
@malqinneh the first link shows blank screens for me - (2nd link works!)

Any idea what's happening?

**Comment by peterbe** [comment]
The way we build the [`UserSubscription` model](https://github.com/mdn/kuma/blob/c658149e815f03217ea0f3621cb8f0d43cece631/kuma/users/models.py#L197) is that a record gets created every time you, basically, finish typing in your credit card on the Stripe SDK. Then, if you cancel your subscription (either via our UI or via Stripe's own dashboard) that record changes to being canceled. If you, for some reason, decide to come back and subscribe again, you'll get a new row in this database table. 

We can compute your membership number by simply counting how many UserSubscription records there exists BEFORE your `created` date. That will pretty much always be the same for every user but it's not a rock solid thing. E.g. `UserSubscription.objects.filter(created__lt=request.user.usersubscription.created).count()`
If, for example, a user (who subscribed before you) decides to delete her account forever, then the count will change for me next time. So I wouldn't want to print *my* count number on a t-shirt. 

So, if it's OK that the number potentially fluctuates (i.e. decreases) if other subscribers later self-destruct, then we can accomplish this user story without any changes to state. 
However, if your subscription membership increment number is to become solid enough that you can engrave it into a rock, then we need to make some new bigger changes. 

To iterate, imagine if I subscribe and I'm subscriber number 100. Then, I immediately get cold feet when I see my credit card statement, so I cancel my subscription. Then, a moment later I realized I looked at the wrong credit card statement and that I *can* indeed afford it so I subscribe again. Then it's going to say that I'll be subscriber number 101.

**Comment by tobinmori** [comment]
Thanks @peterbe. I had in my mind that it would be static (engraveable on a rock). And therefore a stored value independent of the rows. 

@malqinneh @atopal - what's your take?

**Comment by malqinneh** [comment]
"To iterate, imagine if I subscribe and I'm subscriber number 100. Then, I immediately get cold feet when I see my credit card statement, so I cancel my subscription. Then, a moment later I realized I looked at the wrong credit card statement and that I can indeed afford it so I subscribe again. Then it's going to say that I'll be subscriber number 101."

I think this is how it should work. Otherwise, there is no incentive to *keep* people from canceling. As a subscriber with a number < 100,000 (or if we create special tier badges or perks later) it becomes harder to simply cancel if I know I will not be able to get back the original subscriber number (which earned me special perks).

**Comment by malqinneh** [comment]
> @malqinneh the first link shows blank screens for me - (2nd link works!)
> 
> Any idea what's happening?

Just updated the links again with dedicated pages inside the collection. Let me know if that works.

**Comment by tobinmori** [comment]
Just to be clear. If this number is dynamically calculated then a user can't go back and check in their acct settings and see her number. Right @peterbe? 

Isn't that contrary to what you want @malqinneh ?

**Comment by malqinneh** [comment]
Can we jump on a call to discuss trade-offs? I'm not confident I understand technical limitation too well to suggest a working solution.

**Comment by peterbe** [comment]
> Just to be clear. If this number is dynamically calculated then a user can't go back and check in her acct settings and see her _same_ number (and rely on it being the same). Right @peterbe?

Correct. With the "stateless" version. If we go the extra mile and write down the number so it never changes, ie. the stateful version, then it won't change. 

One is more complex. One is simple. Or, one doesn't belong in an MVP the other does. Does that help?

**Comment by malqinneh** [comment]
The concept of assigning a unique number to subscribers is so that they can feel ownership over that number. In the scenario you're talking about, subscriber #100 can one work their way to be subscriber #1. Is that correct?

**Comment by peterbe** [comment]
> The purpose of assigning a unique number to subscribers is so that they can feel ownership over that number. In the scenario you're describing, subscriber 100 can one work their way to be subscriber 1. Is that correct?

Yeah, if you can convince the other 99 people, who became subscribers before you, to delete their accounts. 

But also, to be clear, in the stateless version, if you were subscriber # 100 and you cancel your subscription, and decide to become a subscriber again, you won't be # 100 this time. It'll be some higher number.

**Comment by tobinmori** [comment]
Correct. There is very likely no ownership over the number, given earlier subscribers may delete their accts/delete subscriptions. 


> The purpose of assigning a unique number to subscribers is so that they can feel ownership over that number. In the scenario you're describing, subscriber 100 can one work their way to be subscriber 1. Is that correct?

Correct. There is very likely no ownership over the number, given earlier subscribers may delete their accts/delete subscriptions. 

So if the point of this feature is Numeric Ownership - this is probably not the way to go. :)
